### ISSUE TITLE
lint: elseif is very noisy (false positives?)

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#420]
Does report this pattern very frequently:
```go
		if f(r) {
			if wasField {
				spans = append(spans, span{start: fromIndex, end: i})
				wasField = false
			}
		} else {
			if !wasField {
				fromIndex = i
				wasField = true
			}
		}
```
There are lots of similar cases under GOROOT.
